### PR TITLE
Ghetto comments updater

### PR DESCRIPTION
**Pull request by andrewconner**
This is a very simple thread updater. It works by hijacking `markAsRead`, which returns how many new messages there are, given message last known ID. When we know there's some new messages, trigger a refetch of the latest batch.

This also refactors the code a bit. `$scope.comments` is all known comments, `$scope.keep.discussion` is no longer modified at all. `$scope.keep.discussion.numMessages` is no longer maintained / updated, because it broke anyways if the number of messages in a discussion changed externally. We now know if we should fetch the next batch based on the initial state, and whether the last batch fetched had messages.

This can become a bit more efficient because `comments` _should_ always be sorted, so we can contextually derive more information. But since the number of objects here is small, we're sorting again on every batch and being naïve.

I'm actually working on a real long-polling implementation, but @leogrim is doing something that makes the long poller way more efficient. So I'm waiting for that until his stuff is done.

@cvializ @camhashemi

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js [hunk 1]
```diff
@@ -49,8 +49,9 @@ angular.module('kifi')
           }
 
           $scope.comments = $scope.keep.discussion.messages.slice().sort(bySentAt); // don't mutate the original array, in case we need it later
+
           $scope.visibleCount = Math.min(3, $scope.comments.length);
-          $scope.showViewPreviousComments = ($scope.visibleCount < $scope.keep.discussion.numMessages);
+          $scope.showViewPreviousComments = $scope.hasMoreToFetch = $scope.visibleCount < $scope.keep.discussion.numMessages;
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:54]: thought `$scope.keep.discussion.numMessages` is being removed

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:54]: Sorry, it's no longer used after rendering. It comes from the server, so on first render, is correct. After that, not used.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/common/directives/comments/comments.js:54]: got it

**Comment by camhashemi** [comment]
looks good to me

**Pull request merged**